packages/utils/src/transactionManager.ts

```diff
@@ -22,6 +22,12 @@ export type VariantTransactionData = {
 };
 export type TransactionData = InvariantTransactionData & VariantTransactionData;
 
+// Used to include *all* info for both sending and receiving crosschain data
+export type CrosschainTransactionData = InvariantTransactionData & {
+  sending: VariantTransactionData;
+  receiving?: VariantTransactionData;
+};
```

Comment:
A concern I have is around putting a requirement of the sdk consumer to know too much about the protocol to understand this type. That being said, I think it is *really* weird that the `amount`, `expiry`, or `preparedBlockNumber` would change on an active transaction once it changes status